puppetry/kinectv2/pygame_button.py

```diff
@@ -0,0 +1,167 @@
+    if PYGLogger is not None:
+        PYGLogger(message)
+
+
+
+
+# Simple button class for pygame_x
+class PYGButton(object):
+
+    def __init__(self, center_x, center_y, width, height, 
+                button_color, hover_color,
+                name, text, text_size, text_color, click_callback = None):
+        """ make a button """
+        self._left = center_x - (width // 2)
+        self._top = center_y - (height // 2)
+        self._width = width
+        self._height = height
+        self._right = self._left + width
+        self._bottom = self._top + height
+        self._color = button_color
+        self._hover_color = hover_color
+        self._name = name
+        self._label = text
+        self._text_size = text_size
+        self._text_color = text_color
+        self._click_callback = click_callback
+        self._mouse_down = False
+
+        self._font = pygame.font.Font("freesansbold.ttf", text_size)
+
+        font_size = self._font.size(self._label)
+        self._text_x = (center_x - (font_size[0] // 1.5))
+        self._text_y = (center_y - (font_size[1] // 2))
+
+
+    def drawNormal(self, surface):
+        """ Draw normal frame, color and text for button """
+        pygame.draw.rect(surface, self._color, (self._left, self._top, self._width, self._height))              
+        rendered_text = self._font.render(F"{self._label}",True, self._text_color)
+        surface.blit(rendered_text, [self._text_x, self._text_y])
+
+
+    def drawHover(self, surface, mouse_down):
+        """ Draw hover frame, color and text for button  TO DO - use mouse_down, add _text_hover_color ?"""
+        pygame.draw.rect(surface, self._hover_color, (self._left, self._top, self._width, self._height))              
+        rendered_text = self._font.render(F"{self._label}",True, self._text_color)
+        shift = 4 if mouse_down else 0  
+        surface.blit(rendered_text, [self._text_x + shift, self._text_y + shift])
+
+
+    def mouseOverButton(self, mouse_pos):
+        """ Return True if mouse is over the button space """
+        mob = mouse_pos[0] > self._left and mouse_pos[0] < self._right and \
+              mouse_pos[1] > self._top and mouse_pos[1] < self._bottom
+        # log_msg("MOB %s : (%r,%r) x (%r, %r), mouse %r over: %r" % 
+        #                 (self._label, self._left, self._top, self._right, self._bottom,
+        #                 mouse_pos, mob))
+        return mob
+
+    def handleMouse(self, surface, mouse_pos, mouse_down):
+        """ Process the given mouse state once per frame """
+        mob = self.mouseOverButton(mouse_pos)
+        if mob:
+            if self._mouse_down and not mouse_down and \
+                self._click_callback is not None:
+                # mouse was down, now up -> click
+                self._click_callback(self._name)
+            self._mouse_down = mouse_down
+            self.drawHover(surface, mouse_down)
+        else:       # Mouse outside

```

Comment:
This needing a comment might be an indication that the variable name `mob` is a bit ambiguous.

Reply:
Understood, I habitually add comments and some are obvious and sometimes obscure.  My creative writing side liked that "Mouse Over Button" can be abbreviated as 'mob'

Reply:
Fine with me, thanks for the consideration.